src/bufr/IodaEncoder/IodaDescription.cpp

```diff
@@ -83,21 +83,34 @@ namespace Ingester
             }
         }
 
-        auto dimConfs = conf.getSubConfigurations(ConfKeys::Dimensions);
-        if (dimConfs.size() == 0)
+        if (conf.has(ConfKeys::Dimensions))
         {
-            std::stringstream errStr;
-            errStr << "ioda::dimensions must contain a list of dimensions!";
-            throw eckit::BadParameter(errStr.str());
-        }
+            auto dimConfs = conf.getSubConfigurations(ConfKeys::Dimensions);
+            if (dimConfs.size() == 0) {
+                std::stringstream errStr;
+                errStr << "ioda::dimensions must contain a list of dimensions!";
+                throw eckit::BadParameter(errStr.str());
+            }
 
-        for (const auto& dimConf : dimConfs)
-        {
-            DimensionDescription scale;
-            scale.name = dimConf.getString(ConfKeys::Dimension::Name);
-            scale.size = dimConf.getString(ConfKeys::Dimension::Size);
+            for (const auto &dimConf : dimConfs) {
+                DimensionDescription dim;
+                dim.name = dimConf.getString(ConfKeys::Dimension::Name);
+
+                if (dimConf.has(ConfKeys::Dimension::Paths))
+                {
+                    dim.paths = dimConf.getStringVector(ConfKeys::Dimension::Paths);
+                }
+                else if (dimConf.has(ConfKeys::Dimension::Path))
+                {
+                    dim.paths = {dimConf.getString(ConfKeys::Dimension::Path)};
+                }
+                else
+                {
+                    throw eckit::BadParameter(R"(ioda::dimensions section must have either "path" or "paths".)");
```

Comment:
R? I guess that allows for the " ?

Reply:
R stands for Raw String. Basically special characters are ignored (not processed) between "( and )".